lib/Products/Product.hs

```diff
@@ -1,17 +1,25 @@
-{-# LANGUAGE OverloadedStrings #-}
-
 module Products.Product where
-  import Control.Monad.Trans.Class (MonadTrans)
-  import Control.Monad.IO.Class (MonadIO)
-  import Database
-  import qualified Database.Persist as DB
+  import qualified Data.Text as T
+  import Database (runDB)
   import qualified Database.Persist.Postgresql as DB
   import GHC.Int (Int64)
   import Models
 
+  data Product = Product { productName :: T.Text 
+                         } deriving (Show)
+
```

Comment:
It does feel like we shouldn't directly be working with `Entity ProductModel` types so I like this idea. It does introduce the need to change our type whenever we add a column but that's not a huge deal. The biggest reason I like it is that it lets you work with associated models so our product could turn into:

```haskell
data Product = Product { productName :: T.Text, productOwner :: User, productFeatures :: [Feature] }
```
although this presents some interesting issues itself.